sdk/src/main/java/com/mapbox/maps/MapboxMap.kt

```diff
@@ -229,7 +229,13 @@ class MapboxMap :
     onMapLoadErrorListener: OnMapLoadErrorListener? = null,
   ) {
     checkNativeMap("loadStyleJson")
-    initializeStyleLoad(onStyleLoaded, onMapLoadErrorListener, styleTransitionOptions)
+    initializeStyleLoad(
+      onStyleLoaded,
+      styleDataStyleLoadedListener = if (styleTransitionOptions != null) {
+        { it.styleTransition = styleTransitionOptions }
+      } else null,
+      onMapLoadErrorListener = styleTransitionOptions
```

Comment:
😄 will ask you later what is this kotlin sorcery

Reply:
Should be `onMapLoadErrorListener = onMapLoadErrorListener`

Reply:
@yunikkk created an [issue](https://youtrack.jetbrains.com/issue/KT-52424/Wrong-smartcast-with-nullable-lambda) to track this btw

Reply:
Updated, the outdated code somehow was still present